server/client.go

```diff
@@ -3164,9 +3177,34 @@ func (c *client) processInboundClientMsg(msg []byte) bool {
 		didDeliver = c.sendMsgToGateways(c.acc, msg, c.pa.subject, c.pa.reply, qnames) || didDeliver
 	}
 
+	// Check to see if we did not deliver to anyone and the client has a reply subject set
+	// and wants notification of no_responders.
+	if !didDeliver && len(c.pa.reply) > 0 {
+		c.mu.Lock()
+		noRespondersRequested := c.opts.NoResponders
+		if noRespondersRequested {
+			if sub := c.subForReply(c.pa.reply); sub != nil {
+				proto := fmt.Sprintf("HMSG %s %s 16 16\r\nNATS/1.0 503\r\n\r\n", c.pa.reply, sub.sid)
+				c.queueOutbound([]byte(proto))
+			}
+		}
+		c.mu.Unlock()
+	}
+
 	return didDeliver
 }
 
+// Return the subscription for this reply subject. Only look at normal subs for this client.
+func (c *client) subForReply(reply []byte) *subscription {
+	r := c.acc.sl.Match(string(reply))
+	for _, sub := range r.psubs {
+		if sub.client == c {
+			return sub
```

Comment:
So we assume there will be a single subscription on the reply subject?

Reply:
No, right now I pick the first from the customer who sent the request. So if there are others doing audits etc this will not select them, has to be from the same client.